app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,98 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(type, id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      resource_type = data['resource_type']
+      collection_type = resource_type.include?('Host') ? 'hosts' : 'vms'
+
+      # The 'auth_user' param must be deleted since the model will otherwise
+      # pass the data hash directly as params to ConversionHost.new.
+      auth_user = data.delete('auth_user')
+      resource = resource_search(data['resource_id'], resource_type, collection_class(collection_type))
```

Comment:
what will happen when this search return nil? what happen if it raise error?
(Just want to know the behavior of api)

Reply:
The api will come back with e.g. `{"error":{"kind":"not_found","message":"Couldn't find Host with 'id'=9999","klass":"ActiveRecord::RecordNotFound"}}`